core/src/main/java/com/spotify/metrics/net/NetstatMetricSet.java

```diff
@@ -0,0 +1,151 @@
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().badSegmentsReceived;
+                }
+            });
+        gauges.put(MetricId.build().tagged("what", "tcp-timeouts", "unit", "timeouts/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().tcpTimeouts;
+                }
+            });
+        gauges.put(MetricId.build().tagged("what", "tcp-lost-retransmit", "unit", "retransmit/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().tcpLostRetransmit;
+                }
+            });
+        gauges.put(MetricId.build().tagged("what", "tcp-syn-retrans", "unit", "retransmit/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().tcpSynRetrans;
+                }
+            });
+
+        return gauges;
+    }
+
+    private static Data readData() {
+        return readData(PROC_NETSTAT_PATH, PROC_SNMP_PATH);
+    }
+
+    @VisibleForTesting
+    static Data readData(final String netstatPath, final String snmpPath) {
+        final Data data = new Data();
+        final Map<String, Long> netstatValues = readFile(netstatPath, "TcpExt:");
+        final Map<String, Long> snmpValues = readFile(snmpPath, "Tcp:");
+
+        data.badSegmentsReceived = snmpValues.getOrDefault("InErrs", 0L);
+        data.segmentsRetransmitted = snmpValues.getOrDefault("RetransSegs", 0L);
+        data.tcpLostRetransmit = netstatValues.getOrDefault("TCPLostRetransmit", 0L);
+        data.tcpSynRetrans = netstatValues.getOrDefault("TCPSynRetrans", 0L);
+        data.tcpTimeouts = netstatValues.getOrDefault("TCPTimeouts", 0L);
+
+        return data;
+    }
+
+    private static Map<String, Long> readFile(final String fileName, final String prefix) {
+        final Map<String, Long> kv = new HashMap<>();
+
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                String[] keys = line.trim().split(SPACES);
+                line = br.readLine();
+                String[] values = line.trim().split(SPACES);
+                if (keys.length == values.length && keys.length > 0 &&
+                    prefix.equals(keys[0].trim()) && prefix.equals(values[0].trim())) {
+                    for (int i = 1; i < keys.length; i++) {
+                        long value = Long.parseLong(values[i]);
+                        kv.put(keys[i], value);
+                    }
+                }
+            }
+        } catch (Exception ex) {
+            return Collections.emptyMap();
```

Comment:
do we want to silently fail? It would suck to have not have any indication of problem of reading these files if you are looking into adding these metrics. Maybe LOG withy level warning?

Reply:
The reason for silently failing is that running locally on anything other than on a *nix system will emit this log. Which will be even more confusing unless we do: https://github.com/spotify/semantic-metrics/pull/130#discussion_r1041989927.